### PR TITLE
[Merged by Bors] - feat: `⨅ i : ι, 0 = 0` in `ℕ`

### PR DESCRIPTION
**Pull request by YaelDillies**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Lattice.lean [hunk 1]
```diff
@@ -66,6 +66,12 @@ theorem iInf_of_empty {ι : Sort*} [IsEmpty ι] (f : ι → ℕ) : iInf f = 0 :=
   rw [iInf_of_empty', sInf_empty]
 #align nat.infi_of_empty Nat.iInf_of_empty
 
+@[simp]
+lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 := by
+  cases isEmpty_or_nonempty ι
+  · exact iInf_empty _
```
Comment (urkud) [inline Mathlib/Data/Nat/Lattice.lean:72]:
```suggestion
  · exact iInf_of_empty _
```

File: Mathlib/Data/Nat/Lattice.lean [hunk 2]
```diff
@@ -66,6 +66,12 @@ theorem iInf_of_empty {ι : Sort*} [IsEmpty ι] (f : ι → ℕ) : iInf f = 0 :=
...
+  · exact ciInf_const
```
Comment (b-mehta) [inline Mathlib/Data/Nat/Lattice.lean:73]:
```suggestion
    (isEmpty_or_nonempty ι).elim (fun h => by simp) (fun h => sInf_eq_zero.2 <| by simp)
```
if you'd like. The proof with `ciInf_const` doesn't seem to work unless it's below the instance defined later.

File: Mathlib/Data/Nat/Lattice.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Nat/Lattice.lean]: I went with Bhavik's proof.

File: Mathlib/Data/Nat/Lattice.lean [hunk 3]
```diff
@@ -66,6 +66,10 @@ theorem iInf_of_empty {ι : Sort*} [IsEmpty ι] (f : ι → ℕ) : iInf f = 0 :=
   rw [iInf_of_empty', sInf_empty]
 #align nat.infi_of_empty Nat.iInf_of_empty
 
+@[simp]
+lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 := by
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/Lattice.lean:70]:
```suggestion
lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 :=
```

File: Mathlib/Data/Nat/Lattice.lean [hunk 4]
```diff
@@ -66,6 +66,10 @@ theorem iInf_of_empty {ι : Sort*} [IsEmpty ι] (f : ι → ℕ) : iInf f = 0 :=
   rw [iInf_of_empty', sInf_empty]
 #align nat.infi_of_empty Nat.iInf_of_empty
 
+@[simp]
+lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 :=
+    (isEmpty_or_nonempty ι).elim (fun h => by simp) (fun h => sInf_eq_zero.2 <| by simp)
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/Lattice.lean:71]:
```suggestion
  (isEmpty_or_nonempty ι).elim (fun h => by simp) (fun h => sInf_eq_zero.2 <| by simp)
```

File: Mathlib/Data/Nat/Lattice.lean [hunk 5]
```diff
@@ -66,6 +66,10 @@ theorem iInf_of_empty {ι : Sort*} [IsEmpty ι] (f : ι → ℕ) : iInf f = 0 :=
   rw [iInf_of_empty', sInf_empty]
 #align nat.infi_of_empty Nat.iInf_of_empty
 
+@[simp]
+lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 :=
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Lattice.lean:70]:
```suggestion
/-- This combines `Nat.iInf_of_empty` with `TODO` -/
@[simp]
lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 :=
```
where `TODO` is the lemma that says `[Nonempty ι] : ⨅ i : ι, x = x`

**Comment by eric-wieser** [comment]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**